src/service/layers-impl.js

```diff
@@ -128,13 +129,14 @@ export class LayoutLayers {
     // the document (the scrolling element) or an element scrolls.
     // This forwards to our scroll-dirty system, and eventually to the scroll
     // listener.
-    this.unlisteners_.push(listen(win.document, 'scroll', event => {
-      const {target} = event;
-      const scrolled = target.nodeType == Node.ELEMENT_NODE
-        ? dev().assertElement(target)
-        : scrollingElement;
-      this.scrolled_(scrolled);
-    }, {capture: true, passive: true}));
+    this.listenForScroll_(win.document);
+
+    // Scroll events do not bubble out of shadow trees.
+    // Strangely, iOS 11 reports that document contains things in a shadow tree.
+    // Any node element, however, doesn't.
```

Comment:
> This comment is confusing because a Document is a Node. Do you mean "element node" instead?

Updated to `Element` instead.

> So document.contains() will return true for shadow elements, but documentElement.contains() doesn't?

Correct. I believe this is a bug in Safari, Chrome doesn't.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=189164